src/get-started/timestamping-tutorial.md

```diff
@@ -0,0 +1,1017 @@
+    fn service_id(&self) -> u16 {
+        TIMESTAMPING_SERVICE
+    }
+
+    fn service_name(&self) -> &'static str {
+        SERVICE_NAME
+    }
+
+    fn state_hash(&self, view: &dyn Snapshot) -> Vec<Hash> {
+        let schema = Schema::new(view);
+        schema.state_hash()
+    }
+
+    fn tx_from_raw(
+        &self,
+        raw: RawTransaction,
+    ) -> Result<Box<dyn Transaction>, StreamStructError> {
+        let tx = TimeTransactions::tx_from_raw(raw)?;
+        Ok(tx.into())
+    }
+
+    fn wire_api(&self, builder: &mut ServiceApiBuilder) {
+        PublicApi::wire(builder);
+    }
+}
+
+```
+
+The code above declares a number of methods:
+
+- `service_id` - returns the ID of the service
+- `service_name` - returns the name of the service
+- `state_hash` - calculates the hash of
+  [the blockchain state](../glossary.md#blockchain-state). The method
+  should return [a vector of hashes](../architecture/services.md#state-hash)
+  of the [Merkelized service tables](../glossary.md#merkelized-indices).
+- `tx_from_raw` - deserialization of transactions processed by the service.
+  If the incoming transaction is built successfully, we put it into
+  a `Box<_>`.
+- `wire_api` - connects the API endpoints of the service to the API of the
+  blockchain.
+
+## Create Service Instance
+
+We define the `ServiceFactory` structure in [src/lib.rs][src/lib.rs]. This
+structure builds an instance of our service.
+
+```rust
+#[derive(Debug, Clone, Copy)]
+pub struct ServiceFactory;
+```
+
+The `ServiceFactory` returns the name of the service and includes the
+`make_service` function which will later allow us to include the new service
+into the `NodeBuilder`.
+
+```rust
+impl fabric::ServiceFactory for ServiceFactory {
+    fn service_name(&self) -> &str {
+        SERVICE_NAME
+    }
+
+    fn make_service(
+        &mut self,
+        _: &fabric::Context,
+    ) -> Box<dyn blockchain::Service> {
+    }
+}
+```
+
+## Create Builder for Exonum Node With Timestamping
+
+As the final step of configuring the backend of our Timestamping Service, we
+add the [src/main.rs][src/main.rs] file, which lets us launch an Exonum node
+with additional services.
+
+In this file we import Exonum, its Configuration service, the Time oracle which
+will provide the current time values and the Timestamping Service itself.
+
+```rust
+fn main() {
+    exonum::helpers::init_logger().unwrap();
+    NodeBuilder::new()
+        .with_service(Box::new(exonum_configuration::ServiceFactory))
+        .with_service(Box::new(exonum_time::TimeServiceFactory))
+        .with_service(Box::new(exonum_timestamping::ServiceFactory))
+        .run();
+}
+```
+
+When called, the `main` function will launch an Exonum node with the
+Configuration, Time and Timestamping services.
+
+The backend part of our Timestamping Service is now ready for use.
+However, in the demo on [Github](timestamping) we have taken it a step further
+and addedd frontend.
```

Comment:
added

Reply:
I am not sure this note is necessary here. I would simply direct the reader to Github to see the frontend and how to launch the demo. The examples of requests may be provided in the corresponding sections above where you describe them.